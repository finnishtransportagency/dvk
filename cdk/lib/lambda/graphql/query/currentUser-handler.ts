import { AppSyncResolverEvent } from 'aws-lambda/trigger/appsync-resolver';
import { CurrentUser } from '../../../../graphql/generated';
import { validateJwtToken } from '../../api/login';
import { log } from '../../logger';

export const handler = async (event: AppSyncResolverEvent<void>): Promise<CurrentUser> => {
  log.info({ headers: event.request.headers }, `currentUser(${event.identity})`);
  const token = event.request.headers['x-iam-accesstoken'];
  const data = event.request.headers['x-iam-data'];
  const jwtToken = await validateJwtToken(token, data || '');
  log.debug('JwtToken: %s', jwtToken);
  return {
    name: jwtToken ? jwtToken['custom:etunimi'] + ' ' + jwtToken['custom:sukunimi'] : '????',
  };
};
