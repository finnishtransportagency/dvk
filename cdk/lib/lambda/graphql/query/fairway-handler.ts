import { AppSyncResolverEvent, AppSyncResolverHandler } from 'aws-lambda/trigger/appsync-resolver';
import { Fairway, QueryFairwayArgs } from '../../../../graphql/generated';
import { log } from '../../logger';
import FairwayDBModel from '../../db/fairwayModel';
import { FairwayService } from '../../api/fairwayService';

const fairwayService = new FairwayService();

export const handler: AppSyncResolverHandler<QueryFairwayArgs, Fairway> = async (event: AppSyncResolverEvent<QueryFairwayArgs>): Promise<Fairway> => {
  log.info(`fairway(${event.arguments.id})`);
  const dbModel = await FairwayDBModel.get(event.arguments.id);
  const apiModel = fairwayService.getFairway(event.arguments.id);
  const fairway: Fairway = {
    id: event.arguments.id,
    name: {
      fi: apiModel.nimiFI,
      sv: apiModel.nimiSV || '',
      en: dbModel?.name || '',
    },
    draft1: apiModel.kulkuSyvyys1,
    draft2: apiModel.kulkuSyvyys2,
    draft3: apiModel.kulkuSyvyys3,
    length: apiModel.pituus,
  };
  return fairway;
};
