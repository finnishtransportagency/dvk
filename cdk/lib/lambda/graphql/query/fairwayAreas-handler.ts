import { AppSyncResolverEvent, AppSyncResolverHandler } from 'aws-lambda';
import axios from 'axios';
import { Area, Fairway, QueryFairwayCardArgs } from '../../../../graphql/generated';
import { getVatuHeaders, getVatuUrl } from '../../environment';
import { log } from '../../logger';

export type AlueVaylaAPIModel = {
  jnro: number;
  status: number;
  linjaus: number;
  mitoitusNopeus?: number;
  mitoitusNopeus2?: number;
};
export type AlueAPIModel = {
  id: number;
  nimi?: string;
  mitoitusSyvays?: number;
  harausSyvyys?: number;
  vertaustaso?: string;
  n2000MitoitusSyvays?: number;
  n2000HarausSyvyys?: number;
  n2000Vertaustaso?: string;
  suunta?: number;
  diaariNumero?: string;
  vahvistusPaivamaara?: string;
  omistaja?: string;
  lisatieto?: string;
  vayla?: AlueVaylaAPIModel[];
  tyyppiKoodi?: string;
  tyyppi?: string;
  merkintalajiKoodi?: number;
  merkintalaji?: string;
  liikennointiStatusKoodi?: number;
  liikennointiStatus?: string;
  liikennointiTyyppiKoodi?: number;
  liikenteenTyyppi?: string;
  liikennointiSuuntaKoodi?: number;
  liikennointiSuunta?: string;
  geometria: object;
};

export const handler: AppSyncResolverHandler<QueryFairwayCardArgs, Area[], Fairway> = async (
  event: AppSyncResolverEvent<QueryFairwayCardArgs, Fairway>
): Promise<Area[]> => {
  log.info(`areas(${event.source.id})`);
  const response = await axios.get(`${await getVatuUrl()}/vaylaalueet?jnro=${event.source.id}`, {
    headers: await getVatuHeaders(),
  });
  const areas = response.data as AlueAPIModel[];
  log.debug('areas: %d', areas.length);
  return areas.map((apiArea) => {
    const area: Area = {
      id: apiArea.id,
      name: apiArea.nimi,
      depth: apiArea.mitoitusSyvays,
      draft: apiArea.harausSyvyys,
      referenceLevel: apiArea.vertaustaso,
      n2000depth: apiArea.n2000MitoitusSyvays,
      n2000draft: apiArea.n2000HarausSyvyys,
      n2000ReferenceLevel: apiArea.n2000Vertaustaso,
      direction: apiArea.suunta,
      journalNumber: apiArea.diaariNumero,
      verificationDate: apiArea.vahvistusPaivamaara,
      owner: apiArea.omistaja,
      additionalInformation: apiArea.lisatieto,
      typeCode: apiArea.tyyppiKoodi,
      type: apiArea.tyyppi,
      notationCode: apiArea.merkintalajiKoodi,
      notation: apiArea.merkintalaji,
      operationStatusCode: apiArea.liikennointiStatusKoodi,
      operationStatus: apiArea.liikennointiStatus,
      operationTypeCode: apiArea.liikennointiTyyppiKoodi,
      operationType: apiArea.liikenteenTyyppi,
      operationDirectionCode: apiArea.liikennointiSuuntaKoodi,
      operationDirection: apiArea.liikennointiSuunta,
      geometry: apiArea.geometria,
    };
    area.fairways = apiArea.vayla?.map((apiFairway) => {
      return {
        fairwayId: apiFairway.jnro,
        line: apiFairway.linjaus,
        status: apiFairway.status,
        sizingSpeed: apiFairway.mitoitusNopeus,
        sizingSpeed2: apiFairway.mitoitusNopeus2,
      };
    });
    return area;
  });
};
