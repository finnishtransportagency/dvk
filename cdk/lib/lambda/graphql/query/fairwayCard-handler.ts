import { AppSyncResolverEvent, AppSyncResolverHandler } from 'aws-lambda/trigger/appsync-resolver';
import { FairwayCard, QueryFairwayCardArgs } from '../../../../graphql/generated';
import { log } from '../../logger';
import FairwayCardDBModel from '../../db/fairwayCardDBModel';
import { FairwayService } from '../../api/fairwayService';

const fairwayService = new FairwayService();

export const handler: AppSyncResolverHandler<QueryFairwayCardArgs, FairwayCard | undefined> = async (
  event: AppSyncResolverEvent<QueryFairwayCardArgs>
): Promise<FairwayCard | undefined> => {
  log.info(`fairwayCard(${event.arguments.id})`);
  const dbModel = await FairwayCardDBModel.get(event.arguments.id);
  if (dbModel) {
    const card: FairwayCard = {
      id: dbModel.id,
      name: {
        fi: dbModel.name?.fi || '',
        sv: dbModel.name?.sv || '',
        en: dbModel.name?.en || '',
      },
      fairways: [],
    };
    for (const fairway of dbModel?.fairways || []) {
      card.fairways.push(fairwayService.mapModelsToFairway(undefined, fairway));
    }
    return card;
  }
  return undefined;
};
