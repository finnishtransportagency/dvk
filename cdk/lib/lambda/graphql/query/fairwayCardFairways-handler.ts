import { AppSyncResolverEvent, AppSyncResolverHandler } from 'aws-lambda';
import { Area, Fairway, FairwayCard, NavigationLine, QueryFairwayCardArgs, RestrictionArea } from '../../../../graphql/generated';
import { log } from '../../logger';
import { AlueAPIModel, fetchVATUByFairwayId, NavigointiLinjaAPIModel, RajoitusAlueAPIModel, VaylaAPIModel } from './vatu';

function mapAPIModelToFairway(apiModel: VaylaAPIModel): Fairway {
  const fairway: Fairway = {
    id: apiModel.jnro,
    name: {
      fi: apiModel.nimiFI || '',
      sv: apiModel.nimiSV || '',
    },
    sizing: {
      additionalInformation: apiModel.lisatieto,
      mareograph: apiModel.mareografi,
      minimumTurningCircle: apiModel.minimiKaantosade,
      minimumWidth: apiModel.minimiLeveys,
      normalTurningCircle: apiModel.normaaliKaantosade,
      normalWidth: apiModel.normaaliLeveys,
      reserveWater: apiModel.varavesi,
    },
    typeCode: apiModel.vaylalajiKoodi,
    type: {
      fi: apiModel.vaylaLajiFI,
      sv: apiModel.vaylaLajiSV,
    },
    area: {
      fi: apiModel.merialueFI,
      sv: apiModel.merialueSV,
    },
    lightingCode: apiModel.valaistusKoodi,
    lighting: {
      fi: apiModel.valaistusFI,
      sv: apiModel.valaistusSV,
    },
    owner: apiModel.omistaja,
    startText: apiModel.alunSeloste,
    endText: apiModel.paatepisteenSeloste,
  };
  fairway.sizingVessels = apiModel.mitoitusalus?.map((vesselModel) => {
    return {
      typeCode: vesselModel.alustyyppiKoodi,
      type: vesselModel.alustyyppi,
      draft: vesselModel.syvays,
      length: vesselModel.pituus,
      width: vesselModel.leveys,
      bodyFactor: vesselModel.runkoTaytelaisyysKerroin,
      size: vesselModel.koko,
    };
  });
  fairway.classifications = apiModel.luokitus?.map((classificationModel) => {
    return {
      type: classificationModel.luokitusTyyppi,
      fairwayClassCode: classificationModel.vaylaluokkaKoodi,
      fairwayClass: {
        fi: classificationModel.vaylaluokkaFI,
        sv: classificationModel.vaylaluokkaSV,
      },
    };
  });
  return fairway;
}

async function getNavigationLineMap(fairwayIds: number[]) {
  const lines = await fetchVATUByFairwayId<NavigointiLinjaAPIModel>(fairwayIds, 'navigointilinjat');
  log.debug('lines: %d', lines.length);
  const lineMap = new Map<number, NavigointiLinjaAPIModel[]>();
  for (const line of lines) {
    for (const lineFairway of line.vayla) {
      if (!lineMap.has(lineFairway.jnro)) {
        lineMap.set(lineFairway.jnro, []);
      }
      lineMap.get(lineFairway.jnro)?.push(line);
    }
  }
  return lineMap;
}

function mapNavigationLines(lines: NavigointiLinjaAPIModel[]) {
  return lines.map((apiLine) => {
    const line: NavigationLine = {
      id: apiLine.id,
      draft: apiLine.mitoitusSyvays,
      depth: apiLine.harausSyvyys,
      referenceLevel: apiLine.vertaustaso,
      n2000draft: apiLine.n2000MitoitusSyvays,
      n2000depth: apiLine.n2000HarausSyvyys,
      n2000ReferenceLevel: apiLine.n2000Vertaustaso,
      direction: apiLine.tosisuunta,
      length: apiLine.pituus,
      additionalInformation: apiLine.lisatieto,
      owner: apiLine.omistaja,
      verificationDate: apiLine.vahvistusPaivamaara,
      journalNumber: apiLine.diaariNumero,
      type: apiLine.tyyppi,
      typeCode: apiLine.tyyppiKoodi,
      geometry: apiLine.geometria,
      fairways: [],
    };
    line.fairways = apiLine.vayla?.map((apiFairway) => {
      return {
        fairwayId: apiFairway.jnro,
        status: apiFairway.status,
        line: apiFairway.linjaus,
      };
    });
    return line;
  });
}

async function getAreaMap(fairwayIds: number[]) {
  const areas = await fetchVATUByFairwayId<AlueAPIModel>(fairwayIds, 'vaylaalueet');
  log.debug('areas: %d', areas.length);
  const areaMap = new Map<number, AlueAPIModel[]>();
  for (const area of areas) {
    for (const areaFairway of area.vayla || []) {
      if (!areaMap.has(areaFairway.jnro)) {
        areaMap.set(areaFairway.jnro, []);
      }
      areaMap.get(areaFairway.jnro)?.push(area);
    }
  }
  return areaMap;
}

function mapAreas(areas: AlueAPIModel[]) {
  return areas.map((apiArea) => {
    const area: Area = {
      id: apiArea.id,
      name: apiArea.nimi,
      draft: apiArea.mitoitusSyvays,
      depth: apiArea.harausSyvyys,
      referenceLevel: apiArea.vertaustaso,
      n2000draft: apiArea.n2000MitoitusSyvays,
      n2000depth: apiArea.n2000HarausSyvyys,
      n2000ReferenceLevel: apiArea.n2000Vertaustaso,
      direction: apiArea.suunta,
      journalNumber: apiArea.diaariNumero,
      verificationDate: apiArea.vahvistusPaivamaara,
      owner: apiArea.omistaja,
      additionalInformation: apiArea.lisatieto,
      typeCode: apiArea.tyyppiKoodi,
      type: apiArea.tyyppi,
      notationCode: apiArea.merkintalajiKoodi,
      notation: apiArea.merkintalaji,
      operationStatusCode: apiArea.liikennointiStatusKoodi,
      operationStatus: apiArea.liikennointiStatus,
      operationTypeCode: apiArea.liikennointiTyyppiKoodi,
      operationType: apiArea.liikenteenTyyppi,
      operationDirectionCode: apiArea.liikennointiSuuntaKoodi,
      operationDirection: apiArea.liikennointiSuunta,
      geometry: apiArea.geometria,
    };
    area.fairways = apiArea.vayla?.map((apiFairway) => {
      return {
        fairwayId: apiFairway.jnro,
        line: apiFairway.linjaus,
        status: apiFairway.status,
        sizingSpeed: apiFairway.mitoitusNopeus,
        sizingSpeed2: apiFairway.mitoitusNopeus2,
        sequenceNumber: apiFairway.vaylaalueenJarjestysNro,
      };
    });
    return area;
  });
}

function mapRestrictionAreas(areas: RajoitusAlueAPIModel[]) {
  return areas.map((apiArea) => {
    const area: RestrictionArea = {
      id: apiArea.id,
      value: apiArea.suuruus,
      presenter: apiArea.esittaja,
      journalNumber: apiArea.diaariNumero,
      verificationDate: apiArea.vahvistusPaivamaara,
      modificationDate: apiArea.muutosPaivamaara,
      startDate: apiArea.alkuPaivamaara,
      endDate: apiArea.loppuPaivamaara,
      status: apiArea.paatosTila,
      source: apiArea.tietolahde,
      location: apiArea.sijainti,
      municipality: apiArea.kunta,
      exception: apiArea.poikkeus,
      geometry: apiArea.geometria,
    };
    area.fairways = apiArea.vayla?.map((apiFairway) => {
      return {
        fairwayId: apiFairway.jnro,
      };
    });
    area.types =
      apiArea.rajoitustyypit?.map((t) => {
        return { code: t.koodi, text: t.rajoitustyyppi };
      }) || [];
    if (apiArea.rajoitustyyppi) {
      area.types.push({ text: apiArea.rajoitustyyppi });
    }
    return area;
  });
}

async function getRestrictionAreaMap(fairwayIds: number[]) {
  const areas = await fetchVATUByFairwayId<RajoitusAlueAPIModel>(fairwayIds, 'rajoitusalueet');
  log.debug('restriction areas: %d', areas.length);
  const areaMap = new Map<number, RajoitusAlueAPIModel[]>();
  for (const area of areas) {
    for (const areaFairway of area.vayla || []) {
      if (!areaMap.has(areaFairway.jnro)) {
        areaMap.set(areaFairway.jnro, []);
      }
      areaMap.get(areaFairway.jnro)?.push(area);
    }
  }
  return areaMap;
}

export const handler: AppSyncResolverHandler<QueryFairwayCardArgs, Fairway[], FairwayCard> = async (
  event: AppSyncResolverEvent<QueryFairwayCardArgs, FairwayCard>
): Promise<Fairway[]> => {
  log.info(`fairwayCardFairways(${event.source.id})`);
  const fairwayMap = new Map<number, Fairway>();
  event.source.fairways.forEach((f) => {
    fairwayMap.set(f.id, f);
  });
  const fairwayIds = event.source.fairways.map((f) => f.id);
  log.debug(`fairwayIds: ${fairwayIds}`);
  const lineMap = await getNavigationLineMap(fairwayIds);
  const areaMap = await getAreaMap(fairwayIds);
  const restrictionAreaMap = await getRestrictionAreaMap(fairwayIds);
  const fairways = await fetchVATUByFairwayId<VaylaAPIModel>(fairwayIds, 'vaylat');
  return fairways.map((apiFairway) => {
    const fairway = fairwayMap.get(apiFairway.jnro);
    log.debug('Fairway: %o', apiFairway);
    return {
      ...mapAPIModelToFairway(apiFairway),
      ...fairway,
      navigationLines: mapNavigationLines(lineMap.get(apiFairway.jnro) || []),
      areas: mapAreas(areaMap.get(apiFairway.jnro) || []),
      restrictionAreas: mapRestrictionAreas(restrictionAreaMap.get(apiFairway.jnro) || []),
    };
  });
};
