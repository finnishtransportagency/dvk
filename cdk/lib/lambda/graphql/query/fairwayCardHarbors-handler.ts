import { AppSyncResolverEvent, AppSyncResolverHandler } from 'aws-lambda';
import { FairwayCard, Harbor, QueryFairwayCardArgs } from '../../../../graphql/generated';
import { getOptionalCurrentUser } from '../../api/login';
import HarborDBModel from '../../db/harborDBModel';
import { mapHarborDBModelToGraphqlType } from '../../db/modelMapper';
import { log } from '../../logger';

export const handler: AppSyncResolverHandler<QueryFairwayCardArgs, Harbor[], FairwayCard> = async (
  event: AppSyncResolverEvent<QueryFairwayCardArgs, FairwayCard>
): Promise<Harbor[]> => {
  log.info(`fairwayCardHarbors(${event.source.id})`);
  const harborIds = event.source.harbors?.map((h) => h.id);
  log.debug(`harborIds: ${harborIds}`);
  const harbors: Harbor[] = [];
  const user = await getOptionalCurrentUser(event);
  for (const id of harborIds || []) {
    const model = await HarborDBModel.getPublic(id);
    if (model) {
      harbors.push(mapHarborDBModelToGraphqlType(model, user));
    }
  }
  return harbors;
};
