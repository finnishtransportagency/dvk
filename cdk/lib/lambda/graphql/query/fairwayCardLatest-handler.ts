import { AppSyncResolverEvent, AppSyncResolverHandler } from 'aws-lambda/trigger/appsync-resolver';
import { FairwayCard, QueryFairwayCardArgs } from '../../../../graphql/generated';
import { log } from '../../logger';
import FairwayCardDBModel from '../../db/fairwayCardDBModel';
import { getPilotPlaceMap, getPilotRoutes, mapFairwayCardDBModelToGraphqlType } from '../../db/modelMapper';
import { getOptionalCurrentUser } from '../../api/login';
import { FeatureCollection, GeoJsonProperties, Geometry } from 'geojson';

export const handler: AppSyncResolverHandler<QueryFairwayCardArgs, FairwayCard | undefined> = async (
  event: AppSyncResolverEvent<QueryFairwayCardArgs>
): Promise<FairwayCard | undefined> => {
  log.info(`fairwayCard(${event.arguments.id})`);
  const pilotMap = await getPilotPlaceMap();
  const pilotRoutes = (await getPilotRoutes()) as FeatureCollection<Geometry, GeoJsonProperties>;
  const dbModel = await FairwayCardDBModel.getLatest(event.arguments.id);
  if (dbModel) {
    const user = await getOptionalCurrentUser(event);
    return mapFairwayCardDBModelToGraphqlType(dbModel, pilotMap, user, pilotRoutes);
  }
  return undefined;
};