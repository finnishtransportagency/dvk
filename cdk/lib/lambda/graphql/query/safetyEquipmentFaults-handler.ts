import { SafetyEquipmentFault } from '../../../../graphql/generated';
import { AppSyncResolverEvent } from 'aws-lambda';
import { TurvalaiteVikatiedotAPIModel } from './vatu';
import { log } from '../../logger';
import { cacheResponse, getFromCache } from '../cache';
import { fetchVATUByApi } from '../../api/axios';

function getKey() {
  return 'safetyequipmentfault-graphql';
}

export const handler = async (event: AppSyncResolverEvent<void>): Promise<SafetyEquipmentFault[]> => {
  log.info(`safetyEquipmentFaults(${event.identity})`);
  const key = getKey();
  try {
    const faults = await fetchVATUByApi<TurvalaiteVikatiedotAPIModel>('vikatiedot');
    log.debug('faults: %d', faults.length);
    const response = faults.map((apiFault) => {
      const fault: SafetyEquipmentFault = {
        id: apiFault.vikaId,
        name: {
          fi: apiFault.turvalaiteNimiFI,
          sv: apiFault.turvalaiteNimiSV,
        },
        equipmentId: apiFault.turvalaiteNumero,
        typeCode: apiFault.vikatyyppiKoodi,
        type: {
          fi: apiFault.vikatyyppiFI,
          sv: apiFault.vikatyyppiSV,
        },
        recordTime: Date.parse(apiFault.kirjausAika),
        geometry: apiFault.geometria,
      };
      return fault;
    });
    await cacheResponse(key, response);
    return response;
  } catch (e) {
    log.error('Getting safety equipment faults failed: %s', e);
    const cacheResponseData = await getFromCache(key);
    if (cacheResponseData.data) {
      log.warn('Returning expired response from s3 cache');
      return JSON.parse(cacheResponseData.data);
    } else {
      throw e;
    }
  }
};
