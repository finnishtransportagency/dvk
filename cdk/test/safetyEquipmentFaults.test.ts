import { mockClient } from 'aws-sdk-client-mock';
import { handler } from '../lib/lambda/graphql/query/safetyEquipmentFaults-handler';
import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
import { createReadStream } from 'fs';
import { mockVoidEvent } from './mocks';

const s3Mock = mockClient(S3Client);

jest.mock('../lib/lambda/environment', () => ({
  getFeatureCacheDurationHours: () => 2,
  getEnvironment: () => 'mock',
  isPermanentEnvironment: () => false,
  getHeaders: () => {},
}));

const faults = [
  {
    vikaId: -2139953035,
    vikatyyppiKoodi: 1022541001,
    turvalaiteNumero: 2,
    turvalaiteNimiFI: 'Kuusinen ylempi',
    turvalaiteNimiSV: 'Kuusinen övre',
    vikatyyppiFI: 'Valo pimeä',
    vikatyyppiSV: 'Ljuset slocknat',
    kirjausAika: '2020-06-07T18:33:04.711000',
    geometria: { type: 'Point', coordinates: [26.9622327773, 60.461115575] },
  },
];

let throwError = false;
jest.mock('../lib/lambda/api/axios', () => ({
  fetchVATUByApi: () => {
    if (throwError) {
      throw new Error('Fetching from VATU api failed');
    }
    return faults;
  },
}));

beforeEach(() => {
  jest.resetAllMocks();
  s3Mock.reset();
  throwError = false;
});

it('should get faults from api', async () => {
  const stream = createReadStream('./test/data/faults.json');
  const expires = new Date();
  expires.setTime(expires.getTime() + 1 * 60 * 60 * 1000);
  s3Mock.on(GetObjectCommand).resolves({ Body: stream, Expires: expires });
  const response = await handler(mockVoidEvent);
  expect(response.length).toBe(1);
  expect(response).toMatchSnapshot();
});

it('should get faults from cache when api call fails', async () => {
  const stream = createReadStream('./test/data/faults.json');
  const expires = new Date();
  expires.setTime(expires.getTime() - 1 * 60 * 60 * 1000);
  s3Mock.on(GetObjectCommand).resolves({ Body: stream, Expires: expires });
  throwError = true;
  const response = await handler(mockVoidEvent);
  expect(response.length).toBe(2);
  expect(response).toMatchSnapshot();
});
