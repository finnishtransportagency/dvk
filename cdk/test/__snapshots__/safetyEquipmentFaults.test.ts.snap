// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get faults from api 1`] = `
[
  {
    "equipmentId": 2,
    "geometry": {
      "coordinates": [
        26.9622327773,
        60.461115575,
      ],
      "type": "Point",
    },
    "id": -2139953035,
    "name": {
      "fi": "Kuusinen ylempi",
      "sv": "Kuusinen övre",
    },
    "recordTime": 1591543984711,
    "type": {
      "fi": "Valo pimeä",
      "sv": "Ljuset slocknat",
    },
    "typeCode": 1022541001,
  },
]
`;

exports[`should get faults from cache when api call fails 1`] = `
[
  {
    "equipmentId": 1700,
    "geometry": {
      "coordinates": [
        26.55766,
        60.31134,
      ],
      "type": "Point",
    },
    "id": -2139943633,
    "name": {
      "fi": "Byskär",
      "sv": "Byskär",
    },
    "recordTime": 1682082750000,
    "type": {
      "fi": "Rikkoutunut",
      "sv": "Söndrig",
    },
    "typeCode": 1022541006,
  },
  {
    "equipmentId": 29508,
    "geometry": {
      "coordinates": [
        23.03142,
        60.31464,
      ],
      "type": "Point",
    },
    "id": -2139946880,
    "name": {
      "fi": "Sokeri 1",
      "sv": "Sokeri 1",
    },
    "recordTime": 1650366855000,
    "type": {
      "fi": "Tuhoutunut/kadonnut",
      "sv": "Förstörd/försvunnen",
    },
    "typeCode": 1022541004,
  },
]
`;
